public/app/features/alerting/unified/hooks/useCombinedRule.ts

```diff
@@ -120,3 +160,142 @@ function getRequestState(
 
   return state;
 }
+
+export function useCombinedRule({ ruleIdentifier }: { ruleIdentifier: RuleIdentifier }): {
+  loading: boolean;
+  result?: CombinedRule;
+  error?: unknown;
+} {
+  const { ruleSourceName } = ruleIdentifier;
+  const dsSettings = getDataSourceByName(ruleSourceName);
+
+  const { dsFeatures, isLoadingDsFeatures } = useDataSourceFeatures(ruleSourceName);
+
+  const {
+    currentData: promRuleNs,
+    isLoading: isLoadingPromRules,
+    error: promRuleNsError,
+  } = alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery({
```

Comment:
Now that we've optimized this view we might enable `refetchOnFocus` and `refetchOnReconnect` – I'll add that as a `TODO`